### DISCUSSION TITLE
Validation not working for custom react select component

### DISCUSSION THREAD

**Discussion by azad-steadybiz** [react-hook-form/react-hook-form#2182] [Q&A] [answered]
I have created a custom react select component named MultiSelect and in some other component named SelectWithText I am usign trhat MultiSelect.

Now I have to created a form using SelectWithText but the validation is not working as expected.

**Comment by bluebill1049** [comment, marked as answer]
can you reproduce that in a CSB?

**Reply by azad-steadybiz** [reply]
> can you reproduce that in a CSB? 

https://codesandbox.io/s/strange-knuth-mkxkg

**Reply by bluebill1049** [reply]
please use Controller, this component doesn't support expose ref of that input either the result of the selection. https://react-hook-form.com/api#Controller take at look at the doc and there are examples with MUI as well.

**Reply by azad-steadybiz** [reply]
In this https://codesandbox.io/s/react-hook-form-custom-input-dyh9q?file=/src/index.js example, they have used without Controller

**Reply by bluebill1049** [reply]
this is outdated, take a look here: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Reply by azad-steadybiz** [reply]
ok 
```
  <Controller
      as={
        <div
          className={`MultiSelect ${className ? className : ""} ${
            changeStyle ? `MultiSelect-customStyle` : ""
          }  ${showBorder ? `MultiSelect-showBorder` : ""} `}
          style={{ width: `${width}px` }}
        >
          <Select
            className={`react-select-container`}
            classNamePrefix="react-select"
            clearable={true}
            closeMenuOnSelect={closeMenuOnSelect}
            components={{ DropdownIndicator, Option, MultiValue }}
            hideSelectedOptions={false}
            isClearable={false}
            isMulti={isMulti}
            isSearchable={false}
            placeholder={placeholder}
            value={selectedOption}
            getOptionLabel={(option) =>
              `${showValue ? option.value : option.label}`
            }
            name={name}
            // onChange={handleChange}
          />
        </div>
      }
      control={control}
      options={options}
      name={name}
    />
```

**Reply by azad-steadybiz** [reply]
It's not working

**Reply by bluebill1049** [reply]
which version of RHF are you using? if you share a codesandbox our life can be much easier, because i can probably open the codesandbox and fix it for you, but just do a simple component (don't dump your whole project).

**Reply by azad-steadybiz** [reply]
Got your point. https://codesandbox.io/s/strange-knuth-mkxkg

Please help me to fix this.

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/loving-jones-qdxjg?file=/src/App.js

**Reply by azad-steadybiz** [reply]
Do it with MultiSelect component, that was my question.

**Reply by bluebill1049** [reply]
<img width="660" alt="Screen Shot 2020-07-13 at 7 30 16 pm" src="https://user-images.githubusercontent.com/10513364/87288881-49345780-c53f-11ea-8d81-372286ca101b.png">
what? I am confused, isn't that a multi select.

**Reply by azad-steadybiz** [reply]
No, I mean to say do it with my MultiSelect component. I am able to do it directly with select, but not able do it via custom Component named MultiSelect. don't get confused with the name.

**Reply by bluebill1049** [reply]
then just matter of passing down the props to the your React-Select, but why not move controller inside your MultiSelect?

**Reply by azad-steadybiz** [reply]
Yes I have done that, but the value are not getting register

**Reply by azad-steadybiz** [reply]
The selected values are not getting register

**Reply by azad-steadybiz** [reply]
It's solved

**Reply by azad-steadybiz** [reply]
One more thing, can I set default values for only one field? I have tried this, it's not working.

**Reply by azad-steadybiz** [reply]
@bluebill1049
